src/promptflow-evals/promptflow/evals/evaluators/chat/__init__.py

```diff
@@ -0,0 +1,189 @@
+
+        **Usage**
+
+        .. code-block:: python
+
+            eval_fn = ChatEvaluator(model_config, deployment_name="gpt-4")
+            conversation = [
+                {"role": "user", "content": "What is the value of 2 + 2?"},
+                {"role": "assistant", "content": "2 + 2 = 4", "context": {
+                    "citations": [{"id": "math_doc.md", "content": "Information about additions: 1 + 2 = 3, 2 + 2 = 4"}]}}
+            ]
+            result = chat_eval(conversation=conversation)
+        """
+
+        # TODO: Need a built-in evaluator for retrieval. It needs to be added to `self._rag_evaluators` collection
+        self._rag_evaluators  = [
+            GroundednessEvaluator(model_config, deployment_name=deployment_name),
+            RelevanceEvaluator(model_config, deployment_name=deployment_name),
+        ]
+        self._non_rag_evaluators = [
+            CoherenceEvaluator(model_config, deployment_name=deployment_name),
+            FluencyEvaluator(model_config, deployment_name=deployment_name),
+        ]
+
+    def __call__(self, *, conversation: List[Dict], parallel=True, **kwargs):
+        """Evaluates chat scenario.
+
+        :param conversation: The conversation to be evaluated. Each turn should have "role" and "content" keys.
+                             "context" key is optional for assistant's turn and should have "citations" key with list of citations.
+        :type conversation: List[Dict]
+        :param parallel: If True, use parallel execution for evaluators. Else, use sequential execution. Default is True.
+        :type parallel: bool
+        :return: The scores for Chat scenario.
+        :rtype: dict
+        """
+
+        self._validate_conversation(conversation)
+
+        # Extract questions, answers and contexts from conversation
+        questions = []
+        answers = []
+        contexts = []
+        for turn_num, each_turn in enumerate(conversation):
+            role = each_turn["role"]
+            if role == "user":
+                questions.append(each_turn["content"])
+            elif role == "assistant":
+                answers.append(each_turn["content"])
+                if "context" in each_turn and "citations" in each_turn["context"]:
+                    citations = json.dumps(each_turn["context"]["citations"])
+                    contexts.append(citations)
+
+        # Select evaluators to be used for evaluation
+        compute_rag_based_metrics = True
+        if len(answers) != len(contexts):
+            safe_message = "Skipping rag based metrics as we need citations or " \
+                            "retrieved_documents in context key of every assistant's turn"
+            logger.warning(safe_message)
+            compute_rag_based_metrics = False
+
+        selected_evaluators = []
+        selected_evaluators.extend(self._non_rag_evaluators)
+        if compute_rag_based_metrics:
+            selected_evaluators.extend(self._rag_evaluators)
```

Comment:
are safety metrics considered as default built-in metrics in chat scenario?

Reply:
Checked with @singankit. It's not of scope at this moment.

Reply:
Content safety metrics are not calculated by default. We want to have a sperarate evaluator for it.